File: solutions/492.construct-the-rectangle/construct-the-rectangle.java
class Solution {
    public int[] constructRectangle(int area) {
        int length = (int)Math.sqrt(area);
        int[] res = new int[2];
        for(; length > 0; length --){
            if(area % length == 0){
                res[0] = area / length;
                res[1] = length;
                break;
            }
        }
        return res;
    }
}

File: solutions/492.construct-the-rectangle/construct-the-rectangle.js
/**
 * @param {number} area
 * @return {number[]}
 */
var constructRectangle = function(area) {
    let sqrt = Math.floor(Math.sqrt(area));
    let width = sqrt;
    while((area % width) !== 0){
        width--;
    }
    return [Math.floor(area / width), width];
    
};